edge-hub/watchdog/linux/src/main.rs

```diff
@@ -0,0 +1,96 @@
+extern crate nix;
+
+use std::{error::Error, thread};
+use std::process::{Command, Stdio, Child};
+use std::sync::{Arc};
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::time::Duration;
+use nix::unistd::Pid;
+use nix::sys::signal::{self, Signal};
+use signal_hook::{iterator::Signals, SIGTERM};
+use tracing::{info};
+use tracing_subscriber;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    init_logging();
+
+    let signals = Signals::new(&[SIGTERM])?;
+    let has_received_sigterm = Arc::new(AtomicBool::new(true));
+    let sigterm_listener = has_received_sigterm.clone();
+    thread::spawn(move || {
+        for _sig in signals.forever() {
+            info!("Received SIGTERM for watchdog");
+            sigterm_listener.store(false, Ordering::Relaxed);
```

Comment:
Relaxed is fine because there is no ordering constraints.
https://doc.rust-lang.org/std/sync/atomic/enum.Ordering.html#variant.Relaxed